cmd/etcd-launcher/main.go

```diff
@@ -242,12 +246,16 @@ func etcdCmd(config *config) []string {
 		fmt.Sprintf("--advertise-client-urls=https://%s.etcd.%s.svc.cluster.local:2379,https://%s:2379", config.podName, config.namespace, config.podIP),
 		fmt.Sprintf("--listen-client-urls=https://%s:2379,https://127.0.0.1:2379", config.podIP),
 		fmt.Sprintf("--listen-metrics-urls=http://%s:2378,http://127.0.0.1:2378", config.podIP),
-		fmt.Sprintf("--listen-peer-urls=http://%s:2380", config.podIP),
-		fmt.Sprintf("--initial-advertise-peer-urls=http://%s.etcd.%s.svc.cluster.local:2380", config.podName, config.namespace),
+		fmt.Sprintf("--listen-peer-urls=https://%s:2380", config.podIP),
+		fmt.Sprintf("--initial-advertise-peer-urls=https://%s.etcd.%s.svc.cluster.local:2380", config.podName, config.namespace),
 		"--trusted-ca-file=/etc/etcd/pki/ca/ca.crt",
 		"--client-cert-auth",
 		"--cert-file=/etc/etcd/pki/tls/etcd-tls.crt",
 		"--key-file=/etc/etcd/pki/tls/etcd-tls.key",
+		"--peer-client-cert-auth",
+		"--peer-cert-file=/etc/etcd/pki/tls/etcd-tls.crt",
+		"--peer-key-file=/etc/etcd/pki/tls/etcd-tls.key",
+		"--peer-trusted-ca-file=/etc/etcd/pki/ca/ca.crt",
```

Comment:
Two points:
* the hardcoded paths should be a part of [resources package](https://github.com/kubermatic/kubermatic/blob/master/pkg/resources/resources.go)
* https://github.com/kubermatic/kubermatic/blob/5fc026617f97b4268e7127777c857f58fd1317c3/pkg/resources/etcd/statefulset.go#L231-L235 points to the different tls key/cert that is intended to be used as peer certificate

Bonus:
Please align [pkg/resources/etcd/statefulset.go](https://github.com/kubermatic/kubermatic/blob/5fc026617f97b4268e7127777c857f58fd1317c3/pkg/resources/etcd/statefulset.go) in regard to existing and to-be-created constants to eliminate hardcoded paths